onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc

```diff
@@ -114,6 +131,9 @@ Status SkipLayerNorm<T, Simplified>::ComputeInternal(OpKernelContext* ctx) const
     }
   }
 
+  const bool skip_broadcasted = (skip->Shape()[0] == 1 || skip->Shape()[skip_dims_size-2] == skip_sequence_length) ? true : false;
```

Comment:
skip->Shape()[skip_dims_size-2] == skip_sequence_length is always true.
I think you want to compare skip_seq_len with input_seq_len here.

const bool skip_broadcasted = skip_dims[0] == 1 || skip_dims_size  == 2